gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -277,28 +297,60 @@ var gpii = fluid.registerNamespace("gpii");
     };
 
     /** Invoked on both "start" and "update" phases - in addition to forwarding to gpii.lifecycleManager.executeActions,
-     * it ensures that the session state snapshot is updated with any results and deletes the "start" action set to
-     * ensure this is executed just once (bug here)
-     * @param toSnapshot {Object} is the structure which will be modified to form the snapshot record (aka. originalSettings) and stashed in sessionState
-     * @param solutionRecord {Object} is not modified and consists of the original transformer return (doesn't really appear necessary)
-     * @param isUpdate {boolean} indicates whether we're doing an update - in which case the originalSettings should not be written based on snapshot
+     * it is responsible for saving the settings that are being set (when the saveSnapshot is true) and storing
+     * the list ofapplied solutions to the sessionState
+     *
+     * @param solutionId {String} the ID of the solution
+     * @param solutionRecord {Object} is not modified and consists of the solution and setings to be applied
+     * @param sessionState {Object} the object holding the state of the system
```

Comment:
This list of parameters seems pretty messed up and doesn't agree with the actual argument - also, you have sessionState twice